tests/unittests/tests-nanocoap/tests-nanocoap.c

```diff
@@ -713,6 +713,32 @@ static void test_nanocoap__options_get_opaque(void)
     TEST_ASSERT_EQUAL_INT(-ENOENT, optlen);
 }
 
+/*
+ * Test adding a path from an unterminated string
+ */
+static void test_nanocoap__add_path_unterminated_string(void)
+{
+    uint8_t buf[_BUF_SIZE];
+    coap_pkt_t pkt;
+    uint16_t msgid = 0xABCD;
+    uint8_t token[2] = {0xDA, 0xEC};
+    char path[16] = "/time";
+    size_t path_len = strlen("/time");
+
+    /* some random non-zero character at the end of /time */
+    path[path_len] = 'Z';
+
+    size_t len = coap_build_hdr((coap_hdr_t *)&buf[0], COAP_TYPE_NON,
+                                &token[0], 2, COAP_METHOD_GET, msgid);
+
+    coap_pkt_init(&pkt, &buf[0], sizeof(buf), len);
+    len = coap_opt_add_chars(&pkt, COAP_OPT_URI_PATH, &path[0], path_len, '/');
+
+    char uri[10] = {0};
+    coap_get_uri_path(&pkt, (uint8_t *)&uri[0]);
+    TEST_ASSERT(0 == strncmp(path, uri, path_len));
+}
```

Comment:
Minor point -- you can use TEST_ASSERT_EQUAL_INT for this comparison.

Also compare the return value from `coap_get_uri_path()` to path_len, but remember that the return value includes the zero terminator.